x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/extractor/ExtractedFieldsDetector.java

```diff
@@ -96,21 +99,13 @@ public ExtractedFields detect() {
         return fields;
     }
 
-    private void removeFieldsUnderResultsField(Set<String> fields) {
-        if (resultsField == null) {
-            return;
-        }
-        checkResultsFieldIsNotPresent();
-        // Ignore fields under the results object
-        fields.removeIf(field -> field.startsWith(resultsField + "."));
-    }
-
     private void checkResultsFieldIsNotPresent() {
         // If the task is restarting we do not mind the index containing the results field, we will overwrite all docs
         if (isTaskRestarting) {
             return;
         }
 
+        String resultsField = config.getDest().getResultsField();
```

Comment:
Is it ok if the `config.getDest()` is `null` (this can be the case for memory estimation endpoint as we do not require providing `dest`).

Reply:
Good point. It is fine because `isTaskRestarting` is `true` when we call it from the memory estimation endpoint. But I'll take this into consideration in the work I'm doing on the _info API.